sentinel-dashboard/src/main/webapp/resources/app/scripts/controllers/gateway/flow.js

```diff
@@ -0,0 +1,232 @@
+var app = angular.module('sentinelDashboardApp');
+
+app.controller('GatewayFlowCtl', ['$scope', '$stateParams', 'GatewayFlowService', 'GatewayApiService', 'ngDialog', 'MachineService',
+  function ($scope, $stateParams, GatewayFlowService, GatewayApiService, ngDialog, MachineService) {
+    $scope.app = $stateParams.app;
+
+    $scope.rulesPageConfig = {
+      pageSize: 10,
+      currentPageIndex: 1,
+      totalPage: 1,
+      totalCount: 0,
+    };
+
+    $scope.macsInputConfig = {
+      searchField: ['text', 'value'],
+      persist: true,
+      create: false,
+      maxItems: 1,
+      render: {
+        item: function (data, escape) {
+          return '<div>' + escape(data.text) + '</div>';
+        }
+      },
+      onChange: function (value, oldValue) {
+        $scope.macInputModel = value;
+      }
+    };
+
+    getMachineRules();
+    function getMachineRules() {
+      if (!$scope.macInputModel) {
+        return;
+      }
+
+      var mac = $scope.macInputModel.split(':');
+      GatewayFlowService.queryRules($scope.app, mac[0], mac[1]).success(
+        function (data) {
+          if (data.code == 0 && data.data) {
+            $scope.rules = data.data;
+            $scope.rulesPageConfig.totalCount = $scope.rules.length;
+          } else {
+            $scope.rules = [];
+            $scope.rulesPageConfig.totalCount = 0;
+          }
+        });
+    };
+    $scope.getMachineRules = getMachineRules;
+
+
+    getApiNames();
+    function getApiNames() {
+      if (!$scope.macInputModel) {
+        return;
+      }
+
+      var mac = $scope.macInputModel.split(':');
+      GatewayApiService.queryApis($scope.app, mac[0], mac[1]).success(
+        function (data) {
+          if (data.code == 0 && data.data) {
+            $scope.apiNames = [];
+
+            data.data.forEach(function (api) {
+              $scope.apiNames.push(api["apiName"]);
+            });
+          }
+        });
+    }
+
+    $scope.intervalUnits = [{val: 0, desc: '秒'}, {val: 1, desc: '分'}, {val: 2, desc: '时'}, {val: 3, desc: '天'}];
+
+    var gatewayFlowRuleDialog;
+    $scope.editRule = function (rule) {
+      $scope.currentRule = angular.copy(rule);
+      $scope.gatewayFlowRuleDialog = {
+        title: '编辑流控规则',
+        type: 'edit',
+        confirmBtnText: '保存'
+      };
+      gatewayFlowRuleDialog = ngDialog.open({
+        template: '/app/views/dialog/gateway/flow-rule-dialog.html',
+        width: 780,
+        overlay: true,
+        scope: $scope
+      });
+    };
+
+    $scope.addNewRule = function () {
+      var mac = $scope.macInputModel.split(':');
+      $scope.currentRule = {
+        grade: 0,
```

Comment:
The default grade should be `1` (QPS)

Reply:
And this field does not take effect due to the missing field of `GatewayRuleEntity`.

Reply:
Default grade has been fixed to `1`(QPS), corresponding grade element in page and entity added.

Reply:
👍